ddtrace/contrib/django/patch.py

```diff
@@ -1,17 +1,550 @@
+            if resolver_match:
+                span.set_tag('django.view', resolver_match.view_name)
+                utils.set_tag_array(span, 'django.namespace', resolver_match.namespaces)
+
+                # Django >= 2.0.0
+                if hasattr(resolver_match, 'app_names'):
+                    utils.set_tag_array(span, 'django.app', resolver_match.app_names)
+
+            if route:
+                span.set_tag('http.route', route)
+
+            # Set HTTP Request tags
+            # Build `http.url` tag value from request info
+            # DEV: We are explicitly omitting query strings since they may contain sensitive information
+            span.set_tag(
+                http.URL,
+                parse.urlunparse(parse.ParseResult(
+                    scheme=request.scheme,
+                    netloc=request.get_host(),  # this will include `host:port`
+                    path=request.path,
+                    params='',
+                    query='',
+                    fragment='',
+                ))
+            )
+
+            response = func(*args, **kwargs)
+
+            # Note: this call must be done after the function call because
+            # some attributes (like `user`) are added to the request through
+            # the middleware chain
+            _set_request_tags(span, request)
+
+            if response:
+                span.set_tag(http.STATUS_CODE, response.status_code)
+                if 500 <= response.status_code < 600:
+                    span.error = 1
+                span.set_tag('django.response.class', func_name(response))
+                if hasattr(response, 'template_name'):
+                    utils.set_tag_array(span, 'django.response.template', response.template_name)
+            return response
+
+
+@with_traced_module
+def traced_template_render(django, pin, wrapped, instance, args, kwargs):
+    """Instrument django.template.base.Template.render for tracing template rendering."""
+    template_name = getattr(instance, 'name', None)
+    if template_name:
+        resource = template_name
+    else:
+        resource = '{0}.{1}'.format(func_name(instance), wrapped.__name__)
+
+    with pin.tracer.trace('django.template.render', resource=resource, span_type=http.TEMPLATE) as span:
+        if template_name:
+            span.set_tag('django.template.name', template_name)
+        engine = getattr(instance, 'engine', None)
+        if engine:
+            span.set_tag('django.template.engine.class', func_name(engine))
+
+        return wrapped(*args, **kwargs)
+
+
+def instrument_view(django, view):
+    """Helper to wrap Django views."""
+
+    # All views should be callable, double check before doing anything
+    if not callable(view) or isinstance(view, wrapt.ObjectProxy):
+        return view
+
+    # Patch view HTTP methods and lifecycle methods
+    http_method_names = getattr(view, 'http_method_names', ('get', 'delete', 'post', 'options', 'head'))
```

Comment:
this might work with `viewset` from `rest_framework` but I'm not sure since you don't need to define those as your method names (but they may just proxy to them?)

Reply:
class inheritance can also mess with things too, depends on when/where we patch.